src/libsyntax/attr.rs

```diff
@@ -312,6 +312,15 @@ pub fn find_export_name_attr(diag: &SpanHandler, attrs: &[Attribute]) -> Option<
     })
 }
 
+pub fn contains_extern_indicator(attrs: &[Attribute]) -> bool {
+    contains_name(attrs, "no_mangle") ||
+        contains_name(attrs, "export_name") ||
+        first_attr_value_str_by_name(attrs, "linkage").map(|v| match &*v {
```

Comment:
My definition is basically... If I have a custom linker setup (whether linker script, or compiling a library to be used by C, etc) and `-C lto` breaks it, that's a bug! The exception being relying on unspecified behaviour, like perhaps naming a mangled symbol.

I can probably safely remove the check for the specific type of linkage (external), at least.